External/ChemDraw/chemdrawreaction.h

```diff
@@ -0,0 +1,63 @@
+//
+//  Copyright (c) 2025, Glysade Inc
+//  All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+//     * Redistributions of source code must retain the above copyright
+//       notice, this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above
+//       copyright notice, this list of conditions and the following
+//       disclaimer in the documentation and/or other materials provided
+//       with the distribution.
+//     * Neither the name of Novartis Institutes for BioMedical Research Inc.
+//       nor the names of its contributors may be used to endorse or promote
+//       products derived from this software without specific prior written
+//       permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+// "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+// LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+// A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+// OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+// LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+// DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+#ifndef RDKIT_CHEMDRAW_REACTION_H
+#define RDKIT_CHEMDRAW_REACTION_H
+
+#include <RDGeneral/export.h>
+#include <GraphMol/RDKitBase.h>
+#include <GraphMol/ChemReactions/Reaction.h>
+#include <string>
+
+namespace RDKit
+{
+namespace v2 {
+//---------------------------------------------------------------------------
+//! \name Chemdraw rxn Support
+///@{
+
+//! Parse text in ChemDraw rxn format into a vector of ChemicalReactions
+RDKIT_RDCHEMDRAWREACTIONLIB_EXPORT std::vector<std::unique_ptr<ChemicalReaction>>
+ChemDrawToChemicalReactions(const std::string &rxnBlock, bool sanitize = false,
+			    bool removeHs = false);
+//! Parse a file in ChemDraw rxn format into a vector of ChemicalReactions
+RDKIT_RDCHEMDRAWREACTIONLIB_EXPORT std::vector<std::unique_ptr<ChemicalReaction>>
+ChemDrawFileToChemicalReactions(const std::string &fileName, bool sanitize = false,
+				bool removeHs = false);
+//! Parse a text stream in ChemDraw rxn format into a vector of ChemicalReactions
+RDKIT_RDCHEMDRAWREACTIONLIB_EXPORT std::vector<std::unique_ptr<ChemicalReaction>>
+ChemDrawDataStreamToChemicalReactions(std::istream &rxnStream,
+				      bool sanitize = false,
+				      bool removeHs = false);
+
+}
```

Comment:
No reaction writer?

Reply:
For a reaction writer we need to get proper depiction coordinates and draw the reaction arrows, this will be easier with @DavidACosgrove's improvements but I deemed it out of the current scope.